src/primitives.rs

```diff
@@ -369,9 +369,10 @@ where
                 .map_err(|ref err| VersionizeError::Deserialize(format!("{:?}", err)))?;
         // Construct the object from the array items.
         // Header(T) fields will be initialized by Default trait impl.
-        let mut object = FamStructWrapper::from_entries(&entries);
+        let mut object = FamStructWrapper::from_entries(&entries)
+            .map_err(|ref err| VersionizeError::Deserialize(format!("{:?}", err)))?;
         // Update Default T with the deserialized header.
-        std::mem::replace(object.as_mut_fam_struct(), header);
+        let _ = std::mem::replace(object.as_mut_fam_struct(), header);
```

Comment:
Can you assign the value directly since you are not doing anything with the previous value from the object?

Reply:
done.